### ISSUE TITLE
After adding random crop the loss becomes nan. It must be a bug in the detectron2.

### ISSUE THREAD

**Issue by KnHuq** [facebookresearch/detectron2#3445]
`

                    def custom_mapper(dataset_dict,dataset_location = "../data/GT/proceesed/COCO/all/"):
                        # Implement a mapper, `similar` to the default DatasetMapper, but with your own customizations
                        dataset_dict = copy.deepcopy(dataset_dict)  # it will be modified by code below
                        image = utils.read_image(os.path.join(dataset_location,dataset_dict["file_name"]), format="BGR")
                        transform_list = [T.Resize((1200,1600)),
                                          T.RandomBrightness(0.1,3),
                                          T.RandomContrast(0.1,2),
                                          T.RandomSaturation(.1,2),T.RandomCrop('absolute',[800,800])]
                        
                        image, transforms = T.apply_transform_gens(transform_list, image)
                        dataset_dict["image"] = torch.as_tensor(image.transpose(2, 0, 1))
                    
                        annos = [utils.transform_instance_annotations(obj, transforms, image.shape[:2]) for obj in dataset_dict.pop("annotations")]
                        instances = utils.annotations_to_instances(annos, image.shape[:2])
                        dataset_dict["instances"] = instances
                        dataset_dict["annotations"] = annos
                        
                        return dataset_dict 
                    
                    class AugTrainer(DefaultTrainer):
                        
                        @classmethod
                        def build_train_loader(cls, cfg):
                            return build_detection_train_loader(cfg, mapper=custom_mapper)

`

**Comment by AKMourato** [comment]
Maybe the loss function has degenerated, try to decrease the learning rate.

**Comment by KnHuq** [comment]
Nope. Doesn't help. The problem lies in the transformation method of detectron2. Rather than using native transformation of detectron2, I am using albumentations library and its working fine now. I request the detectron2 team to look at it. Thanks.

**Comment by ppwwyyxx** [comment]
We'll look at it if there is runnable code to show there is a bug.
